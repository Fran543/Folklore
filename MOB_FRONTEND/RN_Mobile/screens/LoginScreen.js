import { View, StyleSheet } from 'react-native';
import AuthContent from '../components/Authentication/AuthContent';
import { Button, Snackbar } from 'react-native-paper';
import * as React from 'react';


export default function LoginScreen() {

    const [visible, setVisible] = React.useState(false);

    const [message, setMessage] = React.useState('');

    const onToggleSnackBar = () => setVisible(!visible);

    const onDismissSnackBar = () => setVisible(false);

    return (
        <View style={styles.container}>
            <AuthContent
                message={message}
                setMessage={(e) => setMessage(e)}
                onToggleSnackBar={onToggleSnackBar} />
            <View style={styles.snackContainer}>
                <Snackbar
                    visible={visible}
                    onDismiss={onDismissSnackBar}
                    action={{
                        label: 'Undo',
                        onPress: () => {
                            // Do something
                        },
                    }}>
                    {message}
                </Snackbar>
            </View>
        </View>
    );
}

const styles = StyleSheet.create({
    snackContainer: {
        justifyContent: 'space-between',
        flex: 1
    },
    container: {
        flex: 1,
        backgroundColor: 'black'
    }
})