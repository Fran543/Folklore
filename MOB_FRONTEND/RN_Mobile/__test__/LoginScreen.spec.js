
import React from 'react';
import renderer from 'react-test-renderer';

import LoginScreen from '../screens/LoginScreen';

describe('<LoginScreen />', () => {
    it('renders correctly', () => {
        const tree = renderer.create(<LoginScreen />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('has 2 child', () => {
        const tree = renderer.create(<LoginScreen />).toJSON();
        expect(tree.children.length).toBe(2);
    });
});