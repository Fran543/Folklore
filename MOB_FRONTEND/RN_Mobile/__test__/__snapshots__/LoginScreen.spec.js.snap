// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoginScreen /> renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "black",
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "#610440",
        "borderRadius": 8,
        "elevation": 2,
        "marginHorizontal": 32,
        "marginTop": 64,
        "padding": 16,
        "shadowColor": "black",
        "shadowOffset": Object {
          "height": 1,
          "width": 1,
        },
        "shadowOpacity": 0.35,
        "shadowRadius": 4,
      }
    }
  >
    <View>
      <View>
        <View
          style={
            Object {
              "marginVertical": 8,
            }
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "white",
                  "marginBottom": 4,
                },
                undefined,
              ]
            }
          >
            Username
          </Text>
          <TextInput
            autoCapitalize="none"
            keyboardType="default"
            onChangeText={[Function]}
            style={
              Array [
                Object {
                  "backgroundColor": "#f9beda",
                  "borderRadius": 4,
                  "fontSize": 16,
                  "paddingHorizontal": 6,
                  "paddingVertical": 8,
                },
                undefined,
              ]
            }
            value=""
          />
        </View>
        <View
          style={
            Object {
              "marginVertical": 8,
            }
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "white",
                  "marginBottom": 4,
                },
                undefined,
              ]
            }
          >
            Email Address
          </Text>
          <TextInput
            autoCapitalize="none"
            keyboardType="email-address"
            onChangeText={[Function]}
            style={
              Array [
                Object {
                  "backgroundColor": "#f9beda",
                  "borderRadius": 4,
                  "fontSize": 16,
                  "paddingHorizontal": 6,
                  "paddingVertical": 8,
                },
                undefined,
              ]
            }
            value=""
          />
        </View>
        <View
          style={
            Object {
              "marginVertical": 8,
            }
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "white",
                  "marginBottom": 4,
                },
                undefined,
              ]
            }
          >
            Password
          </Text>
          <TextInput
            autoCapitalize="none"
            onChangeText={[Function]}
            secureTextEntry={true}
            style={
              Array [
                Object {
                  "backgroundColor": "#f9beda",
                  "borderRadius": 4,
                  "fontSize": 16,
                  "paddingHorizontal": 6,
                  "paddingVertical": 8,
                },
                undefined,
              ]
            }
            value=""
          />
        </View>
        <View
          style={
            Object {
              "marginVertical": 8,
            }
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "white",
                  "marginBottom": 4,
                },
                undefined,
              ]
            }
          >
            Confirm Password
          </Text>
          <TextInput
            autoCapitalize="none"
            onChangeText={[Function]}
            secureTextEntry={true}
            style={
              Array [
                Object {
                  "backgroundColor": "#f9beda",
                  "borderRadius": 4,
                  "fontSize": 16,
                  "paddingHorizontal": 6,
                  "paddingVertical": 8,
                },
                undefined,
              ]
            }
            value=""
          />
        </View>
        <View
          style={
            Object {
              "marginTop": 12,
            }
          }
        >
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Array [
                Object {
                  "backgroundColor": "#c30b64",
                  "borderRadius": 6,
                  "elevation": 2,
                  "paddingHorizontal": 12,
                  "paddingVertical": 6,
                  "shadowColor": "black",
                  "shadowOffset": Object {
                    "height": 1,
                    "width": 1,
                  },
                  "shadowOpacity": 0.25,
                  "shadowRadius": 4,
                },
                false,
              ]
            }
          >
            <View>
              <Text
                style={
                  Object {
                    "color": "white",
                    "fontSize": 16,
                    "fontWeight": "bold",
                    "textAlign": "center",
                  }
                }
              >
                Sign up
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "marginTop": 8,
        }
      }
    >
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "paddingHorizontal": 12,
              "paddingVertical": 6,
            },
            false,
          ]
        }
      >
        <View>
          <Text
            style={
              Object {
                "color": "#f9beda",
                "textAlign": "center",
              }
            }
          >
            Log in instead
          </Text>
        </View>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "flex": 1,
        "justifyContent": "space-between",
      }
    }
  />
</View>
`;
